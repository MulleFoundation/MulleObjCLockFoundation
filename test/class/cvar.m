#import <MulleObjC/MulleObjC.h>


@interface Foo : NSObject
@end


@implementation Foo
@end


@interface NSConstantString : NSObject
@end


@implementation NSConstantString
@end




main()
{
   Class   cls;
   Foo     *foo;
   Foo     *bar;
   NSConstantString  *key1;
   NSConstantString  *key2;

   cls  = [Foo class];
   foo  = [Foo new];
   key1 = [NSConstantString new];  // total fake
   key2 = [NSConstantString new];  // total fake


   bar = [cls classValueForKey:(void *) key1];
   printf( "%s\n", ! bar ? "absent" : "present"),
   bar = [cls classValueForKey:(void *) key2];
   printf( "%s\n", ! bar ? "absent" : "present"),

   [cls setClassValue:foo
               forKey:key1];

   bar = [cls classValueForKey:(void *) key1];
   printf( "%s\n", bar != foo ? "absent" : "present"),
   bar = [cls classValueForKey:(void *) key2];
   printf( "%s\n", bar != foo ? "absent" : "present"),

   [cls removeClassValueForKey:(void *) key1];

   bar = [cls classValueForKey:(void *) key1];
   printf( "%s\n", ! bar ? "absent" : "present"),
   bar = [cls classValueForKey:(void *) key1];
   printf( "%s\n", ! bar ? "absent" : "present"),

   // should be reclaimed by +dealloc
   [cls setClassValue:foo
               forKey:key2];

   [foo release];
   [key1 release];
   [key2 release];
}
