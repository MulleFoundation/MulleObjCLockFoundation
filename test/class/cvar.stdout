absent
absent
present
absent
absent
absent
